lib/evmone/tracing.cpp

```diff
@@ -0,0 +1,64 @@
+// evmone: Fast Ethereum Virtual Machine implementation
+// Copyright 2021 The evmone Authors.
+// SPDX-License-Identifier: Apache-2.0
+
+#include "tracing.hpp"
+#include <evmc/hex.hpp>
+
+namespace evmone
+{
+namespace
+{
+/// @see create_histogram_tracer()
+class HistogramTracer : public Tracer
+{
+    const uint8_t* m_code = nullptr;
+    int32_t m_depth = -1;
+    const char* const* m_opcode_names = nullptr;
+    uint32_t m_counts[256]{};
+    std::ostream& m_out;
+
+    void on_execution_start(
+        evmc_revision rev, const evmc_message& msg, bytes_view code) noexcept override
+    {
+        m_code = code.data();
+        m_depth = msg.depth;
+        if (m_depth == 0)
+            m_opcode_names = evmc_get_instruction_names_table(rev);
```

Comment:
Why save it in the class, and not just call `evmc_get_instruction_names_table` in `on_execution_end` ?

Reply:
This needs revision. So I need save one of these.